src/components/heading/heading.scss

```diff
@@ -0,0 +1,125 @@
+@use '../../design-tokens/colors.yml';
+@use '../../design-tokens/layout.yml';
+@use "../../design-tokens/motion.yml";
+@use '../../design-tokens/opacity.yml';
+@use '../../design-tokens/sizes.yml';
+@use '../../mixins/focus';
+@use '../../mixins/headings';
+@use '../../mixins/ms';
+@use '../../mixins/unit';
+@use 'sass:math';
+
+$permalink-icon-size: sizes.$control-icon-size;
+$permalink-radius: sizes.$radius-full;
+$permalink-size: sizes.$control-height;
+
+$min-width-permalink-shift: layout.$max-width-prose + $permalink-size * 4;
+$max-width-permalink-shift: ($min-width-permalink-shift * 16 - 1) / 16;
+
+/**
+ * This containing element may be applied to a heading directly to apply a
+ * heading style, or to a containing element if you want to display a heading
+ * and permalink.
+ *
+ * 1. We default to the lowest heading level supported by HTML5, which forces
+ *    the template designer to specify an intended visual heading level.
+ * 2. Sets up the optional permalink to position itself relative to the
+ *    containing element.
+ */
+
+.c-heading {
+  @include headings.level(6); /* 1 */
+  position: relative; /* 2 */
+}
+
+/**
+ * Level modifiers
+ *
+ * These will apply the styles for a specific heading level.
+ *
+ * We start at `-2` so template designers have three sizes larger than a default
+ * `<h1>` to apply thoughtfully within templates.
+ *
+ * We end at 3 because that is the maximum depth we support with distinct
+ * visual differences.
```

Comment:
Great question! I hadn't, so I did an export of our blog articles to see. Here's what I found:

Heading | Occurrences | Before 2016 Redesign | After 2016 Redesign
--- | --- | --- | ---
h1 | 0 | 0 | 0
h2 | 369 | 93 | 276
h3 | 616 | 485 | 131
h4 | 222 | 201 | 21
h5 | 23 | 23 | 0
h6 | 0 | 0 | 0

The reason the "before" and "after" our redesign is important is that, prior to that change, the `<h2>` element would result in a heading with the same visual hierarchy as the article title. Here's a screenshot of that design with a heading changed to `<h2>` so you can see:

<img width="609" alt="Screen Shot 2020-05-04 at 10 17 51 AM" src="https://user-images.githubusercontent.com/69633/80993879-b1084c80-8df0-11ea-9e1c-d165ff43eb34.png">

When our new design freed up `<h2>` for use inside of articles, you can see our heading usage shifted one level up.

To answer your original question, `<h4>` elements seem to be our deepest and level since 2016. I think that supports it being more subtly differentiated than the higher headings.